tide-developing-a-distributed-database-in-a-breeze.md

```diff
@@ -0,0 +1,117 @@
+
+So, to welcome newcomers to TiDB and make it easier for them to contribute to our community, we've developed a TiDB integrated development environment: [TiDE](https://github.com/tidb-incubator/tide). Created during [TiDB Hackathon 2020](https://pingcap.com/community/events/hackathon2020/), **TiDE is a Visual Studio Code extension that makes developing TiDB a breeze**. With [this extension](https://marketplace.visualstudio.com/items?itemName=dragonly.ticode), developing a distributed system can be as easy as developing a local one.
+
+In this post, **I'll show you the core features of TiDE and how it smoothes out the standard working process for TiDB developers**. Let's get started.
+
+## How hard was it to develop TiDB?
+
+As a complicated distributed SQL database system, [TiDB](https://github.com/pingcap/tidb) has three core components (TiDB, TiKV, and the Placement Driver, PD) and many ecosystem tools. TiDB and [PD](https://github.com/tikv/pd) are written in Go, whereas [TiKV](https://github.com/tikv/tikv) is written in Rust.
+
+![TiDB architecture in a glance](media/tide-tidb-architecture-in-a-glance.jpg)
+<div class="caption-center"> TiDB architecture in a glance </div>
+
+Imagine you want to get your hands dirty and compile TiDB all by yourself. You need to go through several painful steps:
+
+1. Download the code of the three components. If you haven't done so already, install Go and Rust toolchains, and compile the binary files according to each component's documentation.
+2. To easily run the compiled component, you need to install [TiUP](https://docs.pingcap.com/tidb/stable/tiup-overview), a package manager for TiDB. You'll then learn to write a basic configuration file and start the TiDB cluster.
+3. After the cluster is up and running, you'll want to read the logs of each component and understand their details by searching the logs for keywords. However, TiDB is a multi-process (and multi-node) system, so you'll need to log in to different physical machines to view logs. You must know how to use `less` and `grep` commands.
+4. When you come up with an idea and want to experiment with TiDB, you need to write your code, compile it, copy the binary files to the local or remote TiUP directory, restart the cluster, and repeat Step 3.
+
+As you can see, the traditional workflow of developing TiDB is a lengthy and tediousdreary process, where most of the effort is wasted in repeated, unpleasant operations.
+
+## What is TiDE, and why do we need it?
+
+**Confucious said: "A craftsman must sharpen his tools to do his job."** That applies to TiDB developers (You!) as well. To reduce tedious work and improve efficiency, automation is a must. TiDE, a VS Code extension, automates your TiDB coding experience inside VS Code.
```

Comment:
```suggestion
**[Confucious](https://en.wikipedia.org/wiki/Confucius) (a Chinese philosopher and politician) said: "A craftsman must sharpen his tools to do his job."** That applies to TiDB developers (You!) as well. To reduce tedious work and improve efficiency, automation is a must. TiDE, a VS Code extension, automates your TiDB coding experience inside VS Code.
```

Reply:
I'm afraid that not all English readers know Confucious, so we'd better introduce him.